src/components/TodoComponents/TodoForm.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import './TodoForm.css';
+
+const initialInputState = { task: '', id: '', completed: false };
+
+class TodoForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = initialInputState;
+    this.taskInputBox = React.createRef();
+    this.submitTodo = props.addTodoFunction;
+    this.clearCompletedTodos = props.clearCompletedTodosFunction;
+  }
+
+  onTaskInputChange = (changeEvent) => {
+    this.setState({
+      task: changeEvent.target.value,
+    });
+  }
+
+  onClearButtonClick = () => {
+    this.clearCompletedTodos()
+  }
+
+  submitTask = () => {
+    if (this.state.task.length) {
+      this.setState({ id: Date.now() });
+      this.submitTodo(this.state);
+      this.clearInput()
+    }
+  }
+
+  clearInput = () => {
+    this.setState(initialInputState);
+  }
+
+  componentDidMount() {
+    // To listen for enter key being pressed
+    this.taskInputBox.current.addEventListener('keydown', this.handleKeydownEvent);
```

Comment:
I think this could also be solved with react's onKeyPress or onKeyDown method inside the form/input. This would require some testing but i believe synthetic events also have access to which button was pressed. Also good that You remember to remove the event listener in React after applying it 👍